jetstream/config.py

```diff
@@ -95,15 +95,71 @@ def _lookup_name(
     raise ValueError(f"Could not locate {klass.__name__} {name}")
 
 
+@attr.s(auto_attribs=True)
+class ParameterDefinition:
+    """
+    * distinct_by_branch * indicates whether value for the parameter
+        needs to be specified for each branch
+        or if the value can be applied across all configuration branches (default: False)
+    """
+
+    name: str  # implicit in configuration
+    friendly_name: str
+    description: Optional[str] = None
+    value: Optional[str] = None
+    distinct_by_branch: Optional[bool] = False
+    default: Optional[Union[str, Dict[str, Any]]] = None
+
+    @classmethod
+    def from_dict(cls, d: Mapping[str, Any]) -> "ParameterDefinition":
+        definition = {
+            "name": d["name"],
+            "friendly_name": d.get("friendly_name"),
+            "description": d.get("description"),
+            "value": d.get("value"),
+            "distinct_by_branch": d.get("distinct_by_branch"),
+            "default": d.get("default"),
+        }
+
+        return cls(**definition)
+
+
+_converter.register_structure_hook(ParameterDefinition, lambda obj, _type: ParameterDefinition.from_dict(obj))
+
+
+@attr.s(auto_attribs=True)
+class ParameterSpec:
+    definitions: Dict[str, ParameterDefinition] = attr.Factory(dict)
+
+    @classmethod
+    def from_dict(cls, d: Mapping[str, Any]) -> "ParameterSpec":
+        params: Dict[str, Any] = {"definitions": dict()}
+
+        if not d:
+            return cls(definitions=dict())
+
```

Comment:
I think `d` is always not `None` at this point?
```suggestion

```

Reply:
> This is looking great. If you want you can also merge these changes and then work on on the `distinct_by_branch` logic in a separate PR to keep the changes smaller. Might make reviewing easier. But up to you.

Yeah, I think that's better. Let me double check if the None check is needed.

Reply:
okay, I removed the check and re-run all tests and everything still passes. This suggests to me that indeed this check was not needed.